src/api/api.ts

```diff
@@ -1 +1,7 @@
-export const BASE_URL = 'https://mate.academy/students-api';
+export const BASE_URL = 'https://mate.academy/students-api/';
+
+export async function getAllUsers() {
+  const response = await fetch(`${BASE_URL}/users`);
+
+  return response.json();
+}
```

Comment:
Multiple issues with files in `/api` folder

There is an inconsistency in how the result is treated.
It makes sense to have generic post<T>(...), get<T>(...), etc methods and use them like so:

```
export const async get<T>(URL: string) => {
   try{
      const response = await fetch(`${BASE_URL}/users`);
 
   // do we need to check ok state ?
// throw or return empty?
 if (!response.ok) {
    throw new Error(`${response.status} - ${response.statusText}`);
  }

  return response.json() as T;

  } catch {
 // handle error?
// allow fail? - if so you need to fix all usages of the API to handle the crash
}
}

export const getComments<Comment[]>(url: string) => {
  return get<Comment[]>('/coments')
}

Reply:
In addition, there are some corner cases as loading post by missing`postId`, etc

Overall - we should always assume that network or backend may fail, but front end should handle it